File: src/m/classes/bamgmesh.js
//BAMGMESH class definition
//
//   Usage:
//      bamgmesh(varargin)

function bamgmesh(){
	//methods
	this.constructor = function(args) {// {{{
		//BAMGMESH - constructor for bamgmesh object
		//
		//   Usage:
		//      bamgmesh = bamgmesh(varargin)

		//initialize list
        switch (args.length) {
            case 0:
				//if no input arguments, create a default object
                break;
            case 1:
                var object = args[0];
                for (var field in object) {
                    if (object.hasOwnProperty(field)) {
                        this[field] = object[field];
                    }
                }
                break;
            default:
				throw Error('bamggeom constructor error message: unknown type of constructor call');
        }
	}// }}}
    this.disp= function(){// {{{
        disp(sprintf('\n%s = \n', 'bamgmesh'));
        disp(this);
    }// }}}

    //properties 
    // {{{
    this.Vertices                   = [];
    this.Edges                      = [];
    this.Triangles                  = [];
    this.IssmEdges                  = [];
    this.IssmSegments               = [];
    this.VerticesOnGeomVertex       = [];
    this.VerticesOnGeomEdge         = [];
    this.EdgesOnGeomEdge            = [];
    this.SubDomains                 = [];
    this.SubDomainsFromGeom         = [];
    this.ElementConnectivity        = [];
    this.NodalConnectivity          = [];
    this.NodalElementConnectivity   = [];
    this.CrackedVertices            = [];
    this.CrackedEdges               = [];
    this.PreviousNumbering          = [];

	this.constructor(arguments);
    //}}}
}

File: src/m/classes/bamgmesh.m
%BAMGMESH class definition
%
%   Usage:
%      bamgmesh(varargin)

classdef bamgmesh
	properties (SetAccess=public) 
		% {{{
		Vertices=[];
		Edges=[];
		Triangles=[];
		IssmEdges=[];
		IssmSegments=[];
		VerticesOnGeomVertex=[];
		VerticesOnGeomEdge=[];
		EdgesOnGeomEdge=[];
		SubDomains=[];
		SubDomainsFromGeom=[];
		ElementConnectivity=[];
		NodalConnectivity=[];
		NodalElementConnectivity=[];
		CrackedVertices=[];
		CrackedEdges=[];
		PreviousNumbering=[];
		% }}}
	end
	methods
		function bg = bamgmesh(varargin)% {{{

		switch nargin
			case 0
				% if no input arguments, create a default object

			case 1

				bg=bamgmesh;
				object=varargin{1};
				fields=fieldnames(object);
				for i=1:length(fields)
					field=fields{i};
					if ismember(field,properties('bamgmesh')),
						bg.(field)=object.(field);
					end
				end

			otherwise
				error('bamgmesh constructor error message: unknown type of constructor call');
			end
		end%}}}
		function display(bm)% {{{
			disp(sprintf('\n%s = \n',inputname(1)));
			disp(struct(bm))
		end%}}}
	end
end
